Compile failed with nightly-2018-04-08-i686-pc-windows-gnu.

```
error[E0432]: unresolved import `proc_macro::TokenNode`
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:22:27
   |
22 | use proc_macro::{Spacing, TokenNode, TokenStream, TokenTree, TokenTreeIter};
   |                           ^^^^^^^^^ no `TokenNode` in the root. Did you mean to use `TokenTree`?

error[E0432]: unresolved import `proc_macro::TokenTreeIter`
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:22:62
   |
22 | use proc_macro::{Spacing, TokenNode, TokenStream, TokenTree, TokenTreeIter};
   |                                                              ^^^^^^^^^^^^^ no `TokenTreeIter` in the root. Did you mean to use `TokenTree`?

   Compiling libflate v0.1.14
error[E0574]: expected struct, variant or union type, found enum `TokenTree`
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:34:14
   |
34 |         Some(TokenTree { span: _, kind: TokenNode::Op(x, Spacing::Alone) })
   |              ^^^^^^^^^ not a struct, variant or union type

error[E0432]: unresolved import `syntax::ast::SpannedIdent`
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\pear_codegen-0.0.12\src\lib.rs:27:56
   |
27 | use syntax::ast::{ItemKind, MetaItem, FnDecl, PatKind, SpannedIdent};
   |                                                        ^^^^^^^^^^^^ no `SpannedIdent` in `ast`

   Compiling flate2 v1.0.1
error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:52:13
   |
52 |         let key = match key.kind {
   |             ^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: all local variables must have a statically known size

error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:58:21
   |
58 |             None => panic!("expected value for {}", key),
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: required by `std::fmt::ArgumentV1::new`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:62:21
   |
62 |         let _ = map.insert(key, value);
   |                     ^^^^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`

error[E0277]: the trait bound `str: std::marker::Sized` is not satisfied
  --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\data-encoding-macro-internal-0.1.1\src\lib.rs:50:19
   |
50 |     let mut map = HashMap::new();
   |                   ^^^^^^^^^^^^ `str` does not have a constant size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: required by `<std::collections::HashMap<K, V>>::new`

error: aborting due to 7 previous errors

Some errors occurred: E0277, E0432, E0574.
For more information about an error, try `rustc --explain E0277`.
error[E0609]: no field `identifier` on type `syntax::ast::PathSegment`
   --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\pear_codegen-0.0.12\src\lib.rs:136:59
    |
136 |             let penultimate = path.segments[num_segs - 2].identifier.name.as_str();
    |                                                           ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `ident`, `parameters`

error[E0609]: no field `identifier` on type `syntax::ast::PathSegment`
   --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\pear_codegen-0.0.12\src\lib.rs:143:48
    |
143 |         let last = path.segments[num_segs - 1].identifier.name.as_str();
    |                                                ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `ident`, `parameters`

error[E0609]: no field `identifier` on type `syntax::ast::PathSegment`
   --> E:\rustdir\registry\src\mirrors.xx.com-27933024a10d4e91\pear_codegen-0.0.12\src\lib.rs:152:40
    |
152 |     let first_ident = path.segments[0].identifier.name.as_str();
    |                                        ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `ident`, `parameters`

error: aborting due to 4 previous errors

Some errors occurred: E0432, E0609.
For more information about an error, try `rustc --explain E0432`.
error: Could not compile `data-encoding-macro-internal`.
warning: build failed, waiting for other jobs to finish...
error: Could not compile `pear_codegen`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/cmp/tests/lib.rs
+++ b/cmp/tests/lib.rs
@@ -47,7 +47,7 @@ fn difference() {
                DecodeError { position: 4, kind: Length });
     assert_eq!(x.from_base64().unwrap(), vec![0, 0]);
     assert_eq!(base64::decode(x).err().unwrap(),
-               base64::DecodeError::InvalidByte(2, b'\n'));
+               base64::DecodeError::InvalidLength);
     let x = b"AAB";
     assert_eq!(BASE64.decode(x).err().unwrap(),
                DecodeError { position: 0, kind: Length });

--- a/lib/macro/internal/src/lib.rs
+++ b/lib/macro/internal/src/lib.rs
@@ -19,20 +19,21 @@ extern crate syn;
 extern crate data_encoding;
 
 #[cfg(not(feature = "stable"))]
-use proc_macro::{Spacing, TokenNode, TokenStream, TokenTree, TokenTreeIter};
+use proc_macro::token_stream::IntoIter;
+#[cfg(not(feature = "stable"))]
+use proc_macro::{TokenStream, TokenTree};
 use std::collections::HashMap;
 #[cfg(feature = "stable")]
-use syn::{Lit, Token, TokenTree};
-#[cfg(feature = "stable")]
 use syn::parse::IResult;
+#[cfg(feature = "stable")]
+use syn::{Lit, Token, TokenTree};
 
 use data_encoding::{BitOrder, Encoding, Specification, Translate, Wrap};
 
 #[cfg(not(feature = "stable"))]
-fn parse_op(tokens: &mut TokenTreeIter, op: char, key: &str) {
+fn parse_op(tokens: &mut IntoIter, op: char, key: &str) {
     match tokens.next() {
-        Some(TokenTree { span: _, kind: TokenNode::Op(x, Spacing::Alone) })
-            if x == op => (),
+        Some(TokenTree::Op(x)) if x.op() == op => (),
         _ => panic!("expected {:?} after {}", op, key),
     }
 }
@@ -45,17 +46,17 @@ fn parse_op<'a>(input: &'a str, op: Token, key: &str) -> &'a str {
 }
 
 #[cfg(not(feature = "stable"))]
-fn parse_map(mut tokens: TokenTreeIter) -> HashMap<String, TokenNode> {
+fn parse_map(mut tokens: IntoIter) -> HashMap<String, TokenTree> {
     let mut map = HashMap::new();
     while let Some(key) = tokens.next() {
-        let key = match key.kind {
-            TokenNode::Term(term) => term.as_str().to_string(),
+        let key = match key {
+            TokenTree::Term(term) => term.as_str().to_string(),
             _ => panic!("expected key got {}", key),
         };
         parse_op(&mut tokens, ':', &key);
         let value = match tokens.next() {
             None => panic!("expected value for {}", key),
-            Some(value) => value.kind,
+            Some(value) => value,
         };
         parse_op(&mut tokens, ',', &key);
         let _ = map.insert(key, value);
@@ -92,13 +93,13 @@ fn parse_map(mut input: &str) -> HashMap<String, Token> {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_string(map: &mut HashMap<String, TokenNode>, key: &str) -> String {
+fn get_string(map: &mut HashMap<String, TokenTree>, key: &str) -> String {
     let node = match map.remove(key) {
         None => return String::new(),
         Some(node) => node,
     };
     let literal = match node {
-        TokenNode::Literal(literal) => literal,
+        TokenTree::Literal(literal) => literal,
         _ => panic!("expected literal for {}", key),
     };
     match syn::parse::string(&literal.to_string()) {
@@ -116,13 +117,13 @@ fn get_string(map: &mut HashMap<String, Token>, key: &str) -> String {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_usize(map: &mut HashMap<String, TokenNode>, key: &str) -> usize {
+fn get_usize(map: &mut HashMap<String, TokenTree>, key: &str) -> usize {
     let node = match map.remove(key) {
         None => return 0,
         Some(node) => node,
     };
     let literal = match node {
-        TokenNode::Literal(literal) => literal,
+        TokenTree::Literal(literal) => literal,
         _ => panic!("expected literal for {}", key),
     };
     match literal.to_string().parse() {
@@ -140,19 +141,17 @@ fn get_usize(map: &mut HashMap<String, Token>, key: &str) -> usize {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_padding(map: &mut HashMap<String, TokenNode>) -> Option<char> {
+fn get_padding(map: &mut HashMap<String, TokenTree>) -> Option<char> {
     let node = match map.remove("padding") {
         None => return None,
         Some(node) => node,
     };
     let literal = match node {
-        TokenNode::Term(term) if term.as_str() == "None" => return None,
-        TokenNode::Literal(literal) => literal,
+        TokenTree::Term(term) if term.as_str() == "None" => return None,
+        TokenTree::Literal(literal) => literal,
         _ => panic!("expected literal for padding"),
     };
-    Some(syn::parse::character(&literal.to_string()).expect(
-        "expected char for padding",
-    ))
+    Some(syn::parse::character(&literal.to_string()).expect("expected char for padding"))
 }
 #[cfg(feature = "stable")]
 fn get_padding(map: &mut HashMap<String, Token>) -> Option<char> {
@@ -165,18 +164,16 @@ fn get_padding(map: &mut HashMap<String, Token>) -> Option<char> {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_bool(map: &mut HashMap<String, TokenNode>, key: &str) -> Option<bool> {
+fn get_bool(map: &mut HashMap<String, TokenTree>, key: &str) -> Option<bool> {
     let node = match map.remove(key) {
         None => return None,
         Some(node) => node,
     };
     let term = match node {
-        TokenNode::Term(term) => term,
+        TokenTree::Term(term) => term,
         _ => panic!("expected literal for padding"),
     };
-    Some(syn::parse::boolean(term.as_str()).expect(
-        "expected bool for padding",
-    ))
+    Some(syn::parse::boolean(term.as_str()).expect("expected bool for padding"))
 }
 #[cfg(feature = "stable")]
 fn get_bool(map: &mut HashMap<String, Token>, key: &str) -> Option<bool> {
@@ -188,7 +185,7 @@ fn get_bool(map: &mut HashMap<String, Token>, key: &str) -> Option<bool> {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_bit_order(map: &mut HashMap<String, TokenNode>) -> BitOrder {
+fn get_bit_order(map: &mut HashMap<String, TokenTree>) -> BitOrder {
     let node = match map.remove("bit_order") {
         None => return BitOrder::MostSignificantFirst,
         Some(node) => node,
@@ -196,12 +193,8 @@ fn get_bit_order(map: &mut HashMap<String, TokenNode>) -> BitOrder {
     let msb = "MostSignificantFirst";
     let lsb = "LeastSignificantFirst";
     match node {
-        TokenNode::Term(term) if term.as_str() == msb => {
-            BitOrder::MostSignificantFirst
-        }
-        TokenNode::Term(term) if term.as_str() == lsb => {
-            BitOrder::LeastSignificantFirst
-        }
+        TokenTree::Term(term) if term.as_str() == msb => BitOrder::MostSignificantFirst,
+        TokenTree::Term(term) if term.as_str() == lsb => BitOrder::LeastSignificantFirst,
         _ => panic!("expected {} or {} for bit_order", msb, lsb),
     }
 }
@@ -211,12 +204,8 @@ fn get_bit_order(map: &mut HashMap<String, Token>) -> BitOrder {
     let lsb = "LeastSignificantFirst";
     match map.remove("bit_order") {
         None => return BitOrder::MostSignificantFirst,
-        Some(Token::Ident(ref ident)) if ident.as_ref() == msb => {
-            BitOrder::MostSignificantFirst
-        }
-        Some(Token::Ident(ref ident)) if ident.as_ref() == lsb => {
-            BitOrder::LeastSignificantFirst
-        }
+        Some(Token::Ident(ref ident)) if ident.as_ref() == msb => BitOrder::MostSignificantFirst,
+        Some(Token::Ident(ref ident)) if ident.as_ref() == lsb => BitOrder::LeastSignificantFirst,
         Some(_) => panic!("expected {} or {} for bit_order", msb, lsb),
     }
 }
@@ -228,13 +217,12 @@ fn check_present<T>(hash_map: &HashMap<String, T>, key: &str) {
 }
 
 #[cfg(not(feature = "stable"))]
-fn get_encoding(mut hash_map: &mut HashMap<String, TokenNode>) -> Encoding {
+fn get_encoding(mut hash_map: &mut HashMap<String, TokenTree>) -> Encoding {
     check_present(&hash_map, "symbols");
     let spec = Specification {
         symbols: get_string(&mut hash_map, "symbols"),
         bit_order: get_bit_order(&mut hash_map),
-        check_trailing_bits: get_bool(&mut hash_map, "check_trailing_bits")
-            .unwrap_or(true),
+        check_trailing_bits: get_bool(&mut hash_map, "check_trailing_bits").unwrap_or(true),
         padding: get_padding(&mut hash_map),
         ignore: get_string(&mut hash_map, "ignore"),
         wrap: Wrap {
@@ -254,8 +242,7 @@ fn get_encoding(mut hash_map: &mut HashMap<String, Token>) -> Encoding {
     let spec = Specification {
         symbols: get_string(&mut hash_map, "symbols"),
         bit_order: get_bit_order(&mut hash_map),
-        check_trailing_bits: get_bool(&mut hash_map, "check_trailing_bits")
-            .unwrap_or(true),
+        check_trailing_bits: get_bool(&mut hash_map, "check_trailing_bits").unwrap_or(true),
         padding: get_padding(&mut hash_map),
         ignore: get_string(&mut hash_map, "ignore"),
         wrap: Wrap {

--- a/lib/src/lib.rs
+++ b/lib/src/lib.rs
@@ -116,7 +116,7 @@
 //! | Input      | `data-encoding` | `rustc`  | `base64` | GNU `base64`  |
 //! | ---------- | --------------- | -------- | -------- | ------------- |
 //! | `AAB=`     | `Trailing(2)`   | `[0, 0]` | `[0, 0]` | `\x00\x00`    |
-//! | `AA\nB=`   | `Length(4)`     | `[0, 0]` | `Err(2)` | `\x00\x00`    |
+//! | `AA\nB=`   | `Length(4)`     | `[0, 0]` | `Length` | `\x00\x00`    |
 //! | `AAB`      | `Length(0)`     | `[0, 0]` | `[0, 0]` | Invalid input |
 //! | `A\rA\nB=` | `Length(4)`     | `[0, 0]` | `Err(1)` | Invalid input |
 //! | `-_\r\n`   | `Symbol(0)`     | `[251]`  | `Err(0)` | Invalid input |

